checkQC/config.py

```diff
@@ -33,16 +35,33 @@ def _get_config_file(config_path):
         :param config_path: path to the config file or None
         :returns: the content of the config file
         """
+        base_path = resources.files(checkQC)
         try:
-            if not config_path:
-                config_path = resource_filename(Requirement.parse('checkQC'), 'checkQC/default_config/config.yaml')
-                log.info("No config file specified, using default config from {}.".format(config_path))
-
-            with open(config_path) as stream:
-                return yaml.safe_load(stream)
+            with resources.as_file(
+                    base_path / "default_config/config_schema.json") as schema_path:
+                with open(schema_path) as schema_file:
+                    schema = json.load(schema_file)
+
+            if config_path:
+                with open(config_path) as config_file:
+                    config = yaml.safe_load(config_file)
+            else:
+                with resources.as_file(
+                        base_path / "default_config/config.yaml") as config_path:
+                    log.info(
+                        "No config file specified, "
+                        f"using default config from {config_path}."
+                    )
+                    with open(config_path) as config_file:
+                        config = yaml.safe_load(config_file)
+            jsonschema.validate(config, schema)
+            return config
         except FileNotFoundError as e:
             log.error("Could not find config file: {}".format(e))
             raise e
+        except jsonschema.exceptions.ValidationError as e:
+            log.exception("")
```

Comment:
I see the point in not logging the exception error twice, but it could be helpful to point the user to the invalid config file, suggestion:
```suggestion
            log.exception(f"Config {config_file} failed schema validation")
```